app/models/ems_refresh/save_inventory_physical_infra.rb

```diff
@@ -58,4 +59,35 @@ def save_asset_details_inventory(parent, hash)
     return if hash.nil?
     save_inventory_single(:asset_details, parent, hash)
   end
+
+  def ipaddress?(hostname)
+    IPAddr.new(hostname)
+    return true
+  rescue
+    return false
+  end
+
+  def resolve_hostname(ipaddress, ems)
+    ems.hostname = "https://#{Resolv.getname(ipaddress)}/"
+    _log.info("EMS ID: #{ems.id}" + " Resolved hostname successfully.")
```

Comment:
Yes, I can

![hostname](https://user-images.githubusercontent.com/12627705/28584802-70fe6c62-7144-11e7-885f-6b68616cd5aa.png)

After the refresh:
![hostname2](https://user-images.githubusercontent.com/12627705/28584803-7116b04c-7144-11e7-8f88-edd985fd14bd.png)